File: javascript01/step05/test06.js
/* 주제: 객체 - 함수를 이용하여 객체를 생성하기
*/

"use strict"

function createScore(name, kor, eng, math) {
  return { // 빈 객체를 만들고 (변수의 )값을 주기
    // 프로퍼티명: 파라미터 값 (실무에서 이렇게 쓴다.)
    name: name,
    kor: kor,
    eng: eng,
    math: math,
    sum: kor + eng + math,
    aver: (kor + eng + math) / 3
  }
}

var score2 = [
  createScore("홍길동1", 100, 100, 100),
  createScore("홍길동2", 90, 90, 90),
  createScore("홍길동3", 80, 80, 80),
  createScore("홍길동4", 70, 70, 70),
  createScore("홍길동5", 60, 60, 60)
]
console.log(score2)
console.log("홍길동3의 총점: " + score2[2].sum)
console.log('-----------------------------------')

File: java01/src/step05/Test06.java
/* 메서드 : 여러개의 값을 리턴하고 싶을 때 2 */
package step05;

public class Test06 {
  
  static class Student {
    String name;
    int kor;
    int eng;
    int math;
    int sum;
    float aver;
  }
  
  // 4) 사용자(개발자)가 정의한 클래스의 객체에 값을 담은 다음 그 객체의 주소를 리턴할 수 있다. 
  public static Student getStudent(int studentNo) {
    // 지정된 번호의 학생 정보를 찾았다고 가정하자. 
    Student s = new Student();
    s.name = "홍길동"; 
    s.kor = 100;
    s.eng = 90;
    s.math = 80;
    s.sum = 100 + 90 + 80;
    s.aver = (100 + 90 + 80) / 3f;
    return s;
  }

  public static void main(String[] args) {
    Student s2 = getStudent(100);
    // s2메모리에 저장된 주소로 찾아가서 name이라는 이름에 저장된 값
    System.out.println(s2.name);
    System.out.println(s2.kor);
    System.out.println(s2.eng);
    System.out.println(s2.math);
    System.out.println(s2.sum);
    System.out.println(s2.aver);

  }

}

